File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    numbers = sorted(numbers, key = lambda x: str(x)*3, reverse=True)
    answer = ''.join(map(str, numbers))
    
    return str(int(answer)) # 00일 경우 0으로 출력되도록 하기 위해 str -> int -> str로 출력

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] numbersString = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; i++){
            numbersString[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(numbersString, (a, b) -> (b+a).compareTo(a+b));
        
        answer = String.join("", numbersString);
        
        if (answer.startsWith("0")){
            answer = "0";
        }
        
        return answer;
    }
}